be/test/exprs/cast_expr_test.cpp

```diff
@@ -2380,4 +2380,109 @@ TEST_F(VectorizedCastExprTest, unsupported_test) {
     ASSERT_FALSE(Expr::create_vectorized_expr(&pool, cast_expr, &expr3, &runtime_state).ok());
 }
 
+TTypeDesc gen_struct_type_desc(const std::vector<TPrimitiveType::type> field_types,
```

Comment:
should be add more sql test case, can be refs: test/sql/test_function/T/test_cast.

and add nested-struct test case, like struct<a Array, b map, c Json, d struct>...